### ISSUE TITLE
Essay problem

### ISSUE THREAD

**Issue by Lxy811** [jkli1998/T-CAR#4]
![Snipaste_2024-03-01_20-56-30](https://github.com/jkli1998/T-CAR/assets/85209124/b076d1c9-70be-4441-99ab-9c4e1d409ffe)
May I ask if a in this formula 12 is obtained according to formula 13?  Can you explain this formula 12?  Formula 12 in the code is softmax operation, right

**Comment by jkli1998** [comment]
> May I ask if a in this formula 12 is obtained according to formula 13? 

Yes.

> Can you explain this formula 12? 

We enhance the attention of our loss to a small number of triplets by controlling the alpha of the different triplets.


> Formula 12 in the code is softmax operation, right

Yes. the softmax operation on the margined r^c_(s,o)

**Comment by Lxy811** [comment]
![微信截图_20240306131425](https://github.com/jkli1998/T-CAR/assets/85209124/8d539a4c-08a1-403c-94eb-43ca336c7f59)
Hello, I would like to ask, is the recurrence result evaluated according to the code command the result of the corresponding T-CAR whose B is ResNeXt-101-FPN in the paper? But the result is a little low, can you ask me what parameters I can adjust for training?

**Comment by jkli1998** [comment]
> ![微信截图_20240306131425](https://private-user-images.githubusercontent.com/85209124/310370639-8d539a4c-08a1-403c-94eb-43ca336c7f59.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk3MDUxOTgsIm5iZiI6MTcwOTcwNDg5OCwicGF0aCI6Ii84NTIwOTEyNC8zMTAzNzA2MzktOGQ1MzlhNGMtMDhhMS00MDNjLTk0ZWItNDNjYTMzNmM3ZjU5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzA2VDA2MDEzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI5ZGNkMjNkN2RkNDRjNWVjMjgxODg1YmJkYWUyNDNhNTJlZDk2YzIyNDYwZWVmODAyNDViMzMwZTMxNGEzNjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.v9qrepseZLHGN1UonkNsZtbyCVT8FFQRLSaBzH04Hro) Hello, I would like to ask, is the recurrence result evaluated according to the code command the result of the corresponding T-CAR whose B is ResNeXt-101-FPN in the paper? But the result is a little low, can you ask me what parameters I can adjust for training?

It seems that at the RELATIONSHIP NMS is not turned on. You can modify the 82th line in maskrcnn_benchmark/config/defaults.py to _C.DATASETS.POST_NMS = True and run this repo again.

**Comment by Lxy811** [comment]
Thank you very much

> > ![微信截图_20240306131425](https://private-user-images.githubusercontent.com/85209124/310370639-8d539a4c-08a1-403c-94eb-43ca336c7f59.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk3MDUxOTgsIm5iZiI6MTcwOTcwNDg5OCwicGF0aCI6Ii84NTIwOTEyNC8zMTAzNzA2MzktOGQ1MzlhNGMtMDhhMS00MDNjLTk0ZWItNDNjYTMzNmM3ZjU5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzA2VDA2MDEzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI5ZGNkMjNkN2RkNDRjNWVjMjgxODg1YmJkYWUyNDNhNTJlZDk2YzIyNDYwZWVmODAyNDViMzMwZTMxNGEzNjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.v9qrepseZLHGN1UonkNsZtbyCVT8FFQRLSaBzH04Hro) Hello, I would like to ask, is the recurrence result evaluated according to the code command the result of the corresponding T-CAR whose B is ResNeXt-101-FPN in the paper? But the result is a little low, can you ask me what parameters I can adjust for training?
> 
> It seems that at the RELATIONSHIP NMS is not turned on. You can modify the 82th line in maskrcnn_benchmark/config/defaults.py to _C.DATASETS.POST_NMS = True and run this repo again.

Thank you very much

**Comment by Lxy811** [comment]
Hello, I would like to ask how the visualized diagrams and scene diagrams in the paper were obtained

**Comment by jkli1998** [comment]
> Hello, I would like to ask how the visualized diagrams and scene diagrams in the paper were obtained

We visualize our scene graph use PPT with the output of T-CAR. As for `bar graph` in our paper, we employ the `seaborn.barplot`.

**Comment by Lxy811** [comment]
> seaborn.barplot

thank you very much

**Comment by Lxy811** [comment]
![358](https://github.com/jkli1998/T-CAR/assets/85209124/e496a1df-557f-4411-8eeb-456e66740282)
Hello, any other suggestions? The result is a little low, can you ask me what parameters I can adjust for training? This is the same result after batch adjustment.
